src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Reflection;
+
+    internal class HeartbeatDefaultPayload
```

Comment:
Why not this payload uses the same Interface (AddHealthProperty() method i meant)used by clients to add payloads?

Reply:
This class is a utility that we call once upon HeartbeatProvider.Initialize, and then never again.

This is a utility class I created to gather all the 'SDK-supplied' default values that appear in the payload. Payload properties that are SDK-supplied must be kept distinct from any user-supplied extended properties.